packages/core/src/RenderingEngine/WSIViewport.ts

```diff
@@ -247,13 +272,10 @@ class WSIViewport extends Viewport {
       getScalarData: () => this.getScalarData(),
       getSpacing: () => metadata.spacing,
       worldToIndex: (point: Point3) => {
-        const canvasPoint = this.worldToCanvas(point);
-        const pixelCoord = this.canvasToIndex(canvasPoint);
-        return [pixelCoord[0], pixelCoord[1], 0] as Point3;
+        return this.worldToIndex(point);
       },
       indexToWorld: (point: Point3) => {
-        const canvasPoint = this.indexToCanvas([point[0], point[1]]);
-        return this.canvasToWorld(canvasPoint);
+        return this.indexToWorld(point);
```

Comment:
This is providing a Point3 coordinate to a point2 requiring function - you might change the signatures to allow either type here, or just see if you can cast it.  You might need to cast it to unknown first.

Reply:
index coordinates are actually Point3 as they include a frame# in the z coordinate, which is not relevant for this case